includes/utils/class-amp-dom-utils.php

```diff
@@ -108,6 +108,18 @@ function( $matches ) {
 			);
 		}
 
+		/*
+		 * Add a pre-HTML5-style declaration of the encoding since libxml<2.8 doesn't recognize
+		 * HTML5's meta charset. See <https://bugzilla.gnome.org/show_bug.cgi?id=655218>.
+		 */
+		if ( version_compare( LIBXML_DOTTED_VERSION, '2.8', '<' ) ) {
+			$document = preg_replace(
+				'#<meta[^>]+charset="([^"]+)"#i',
```

Comment:
It looks like this may be missing the trailing `>` so the replacement would seem to result in doubled `>>` being added?